src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/IUpgradeContext.cs

```diff
@@ -41,5 +44,7 @@ public interface IUpgradeContext : IDisposable
         ValueTask ReloadWorkspaceAsync(CancellationToken token);
 
         IUpgradeContextProperties Properties { get; }
+
+        ICollector<AnalyzeResultDefinition> UpgradeResults { get; }
```

Comment:
```suggestion
        ICollector<AnalyzeResultDefinition> Results { get; }
```

It's already in the `IUpgradeContext` so it seems a bit redundant

Reply:
Yes. This PR is not yet ready for review. I just wanted to discuss the basic design. This will be cleaned up in subsequent commits